subprojects/core/src/main/java/org/gradle/api/internal/file/DefaultSourceDirectorySet.java

```diff
@@ -164,17 +170,33 @@ public PatternFilterable getFilter() {
 
     @Override
     public File getOutputDir() {
-        return outputDir.get();
+        return destinationDirectory.getAsFile().get();
```

Comment:
I think we should emit a deprecation warning for getOutputDir so people switch to classesDirectory.  This will guarantee that they get the right dependency vs however they're figuring it out now

Reply:
Of course. Totally missed this. Will be part of the follow-up deprecation PR.